Festivity/TicketBuy.cs

```diff
@@ -27,32 +30,79 @@ private static void UpdateCurrentFestivalTickets(int festivalId)
             }
         }
 
-        public static void TicketShow()
+        public static bool AgeCheck(int festivalId) //Checks if the user is old enough to use the program
         {
-            currentFestivalId = CatalogPage.selectedFestival;
-
-            UpdateCurrentFestivalTickets(currentFestivalId);
-
-            MenuFunction.option = 0;
+            foreach (var festival in festivals.Festivals)
+            {
+                if (festival.FestivalID == festivalId)
+                {
+                    foreach (var user in users.Users)
+                    {
+                        if (LoggedInAccount.GetID() == user.AccountID)
+                        {
+                            int userAgeYear = festival.FestivalDate.Year - user.birthDate.Year;
+                            int userAgeMonth = festival.FestivalDate.Month - user.birthDate.Month;
+                            int userAgeDay = festival.FestivalDate.Day - user.birthDate.Day;
+                            if (userAgeYear * 365 + userAgeMonth * 30 + userAgeDay > festival.FestivalAgeRestriction * 365)
+                            {
+                                return true;
+                            }
+                            else
+                            {
+                                return false;
+                            }
+                        }
+                    }
+                }
+            }
+            return true;
+        }
 
-            while (true)
+        public static void TicketShow()
+        {
+            foreach (var festival in festivals.Festivals)
             {
-                List<string> menuOptionsList = new List<string>();
-                string line = "----------------------------------------------------------------------";
+                currentFestivalId = CatalogPage.selectedFestival;
 
-                // Displays the Tickets for the current Festival
-                foreach (var ticket in CurrentTicketList)
+                if (currentFestivalId == festival.FestivalID)
                 {
-                    Console.WriteLine(ticket.TicketName);
-                    Console.WriteLine("Description: " + ticket.TicketDescription);
-                    Console.WriteLine("Price: " + ticket.TicketPrice + " euros");
-                    Console.WriteLine(line);
-                    menuOptionsList.Add("Buy Ticket:" + ticket.TicketID);
+                    if (AgeCheck(currentFestivalId))
+                    {
+                        UpdateCurrentFestivalTickets(currentFestivalId);
+
+                        MenuFunction.option = 0;
+
+                        while (true)
+                        {
+                            List<string> menuOptionsList = new List<string>();
+                            string line = "----------------------------------------------------------------------";
```

Comment:
Duplicate code is bad 😜

Reply:
??

Reply:
I legit don't know which code you're referring to.

If you're talking about "if (currentFestivalId == festival.FestivalID)" (line 69)
This is needed for "festival.FestivalAgeRestriction" (line 100)

Reply:
Fixed